internal/common/pid_sock_status.go

```diff
@@ -24,16 +27,23 @@ func GetStatusPath() string {
 	return filepath.Join(config.GetWorkDir(), STATUS_FILENAME)
 }
 
-func WriteAndHandleStatusFile() error {
+func WriteAndHandleStatusFile() (err error) {
 	Lock(PID_SOCK_LOCK)
-	defer Unlock(PID_SOCK_LOCK)
+	defer func() {
+		err = errors.Join(err, Unlock(PID_SOCK_LOCK))
```

Comment:
No need to handle errors on the Unlock().

You know how strict I am about errors,
but Unlock() is one of the cases that an error would be so unheard of
(and only if we messed up),
that it's fine to skip.